torch/ao/quantization/fx/_convert_do_not_use.py

```diff
@@ -190,12 +202,55 @@ def replace_observer_with_quantize_dequantize_node(graph: Graph, node: Node, mod
                 if isinstance(maybe_dequantize_node, Node) and \
                    maybe_dequantize_node.op == "call_method" and \
                    maybe_dequantize_node.target == "dequantize":
-                    quantized_node = maybe_dequantize_node.args[0]
-                    maybe_dequantize_node.replace_all_uses_with(quantized_node)
+                    quantize_node = maybe_dequantize_node.args[0]
+                    maybe_dequantize_node.replace_all_uses_with(quantize_node)
                     model.graph.erase_node(maybe_dequantize_node)
         elif node.op == "call_module":
             if is_activation_post_process(modules[node.target]):
                 replace_observer_with_quantize_dequantize_node(model.graph, node, modules)
+            elif is_observed_standalone_module(modules[node.target]):
+                # TODO: move this to a separate function
+                convert = torch.ao.quantization._quantize_fx_do_not_use._convert_do_not_use  # type: ignore[attr-defined]
+                # We know that observed standalone module is a GraphModule since
+                # it's produced by us
+                observed_standalone_module : GraphModule = modules[str(node.target)]  # type: ignore[assignment]
+                sm_input_quantized_idxs = \
+                    observed_standalone_module \
+                    ._standalone_module_input_quantized_idxs\
+                    .tolist()  # type: ignore[operator]
+                # remove the dequantize nodes for inputs
+                args = list(node.args)
+                for idx in range(len(args)):
+                    if idx in sm_input_quantized_idxs:
+                        arg = args[idx]
+                        if arg.op == "call_method" and arg.target == "dequantize":
```

Comment:
If the input index is already quantized, why do we expect to have a `dequantize` here for the input index?

Reply:
oh, this is because the new convert flow will output reference pattern by default, and all operators in this flow are assumed to take fp32 inputs, so we'd need to dequantize the input here so that it can be consumed by the following operator

Reply:
I see, so the `quantize_node` on line 227 is not really a `quantize_per_tensor` node, but a node with quantized value?

Reply:
right it's a bit confusing...it actually means both, e.g.
quantize_node means the following node:
```
quantized = call_function[target=torch.quantize_per_tensor](input)
```
but we can also use quantize_node as an argument (value):
```
dequantized = call_method[target="dequantize"](quantized)
```
it comes from the fact that Node in fx means both the operator and the output value, there is no separate value concept in fx graph I think. This also causes some other confusion in our code base.
cc @vkuzo @jamesr66a for any additional comments